components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2968,22 +2968,35 @@ def __init__(self, descriptor, toJSONMethod):
     def definition_body(self):
         ret = ''
         interface = self.descriptor.interface
+
         for m in interface.members:
             if m.isAttr() and not m.isStatic() and m.type.isJSONType():
                 name = m.identifier.name
+                conditions = MemberCondition(None, None, m.exposureSet)
+                ret_conditions = 'vec![' + ",".join(conditions) + "]"
                 ret += fill(
                     """
-                    rooted!(in(cx) let mut temp = UndefinedValue());
-                    if !get_${name}(cx, obj, this, JSJitGetterCallArgs { _base: temp.handle_mut().into() }) {
-                      return false;
-                    }
-                    if !JS_DefineProperty(cx, result.handle().into(),
-                                          ${nameAsArray} as *const u8 as *const libc::c_char,
-                                          temp.handle(), JSPROP_ENUMERATE as u32) {
-                      return false;
+                    let incumbent_global = GlobalScope::incumbent().expect("no incumbent global");
+                    let global = incumbent_global.reflector().get_jsobject();
```

Comment:
We can move these to the initial definition of `ret` so that we only get the global object once at the start of the method.

Reply:
Thanks for the review. fixed 74995a5287dc0af8afe62e9b537db5141b854765